.github/actions/cache-restore/action.yml

```diff
@@ -0,0 +1,40 @@
+name: "Cache restoration"
+description: "Setup a NodeJS environment and restore cache if any, given a node version"
+
+inputs:
+  node-version:
+    description: "Node version to use, default to LTS Fermium (14.x)"
+    required: true
+    default: lts/fermium
+outputs:
+  cache-hit:
+    description: "Forward actions/cache cache-hit output"
+    value: ${{ steps.node-cache.outputs.cache-hit }}
+
+runs:
+  using: "composite" # Mandatory parameter
+  steps:
+    # Setup a Node environment given a node version
+    - name: Use Node.js ${{ inputs.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node-version }}
+
+    - name: Cache Node Modules
```

Comment:
maybe you can try to use this caching mechanism:
https://github.com/actions/setup-node#caching-global-packages-data